tests/unit/lms/resources/_js_config_test.py

```diff
@@ -0,0 +1,161 @@
+
+    def test_it_is_None_for_non_lti_users(
+        self, authToken, BearerTokenSchema, context, pyramid_request
+    ):
+        pyramid_request.lti_user = None
+
+        assert JSConfig(context, pyramid_request).config["authToken"] is None
+
+    @pytest.fixture
+    def authToken(self, config):
+        return config["authToken"]
+
+
+class TestJSConfigHypothesisClient:
+    """Unit tests for the "hypothesisClient" sub-dict of JSConfig."""
+
+    def test_it_contains_one_service_config(self, config):
+        assert len(config["services"]) == 1
+
+    def test_it_includes_the_api_url(self, config):
+        assert config["services"][0]["apiUrl"] == "https://example.com/api/"
+
+    def test_it_includes_the_authority(self, config):
+        assert config["services"][0]["authority"] == "TEST_AUTHORITY"
+
+    def test_it_disables_share_links(self, config):
+        assert config["services"][0]["enableShareLinks"] is False
+
+    def test_it_includes_grant_token(self, config):
+        before = int(datetime.datetime.now().timestamp())
+
+        grant_token = config["services"][0]["grantToken"]
+
+        claims = jwt.decode(
+            grant_token,
+            algorithms=["HS256"],
+            key="TEST_JWT_CLIENT_SECRET",
+            audience="example.com",
+        )
+        after = int(datetime.datetime.now().timestamp())
+        assert claims["iss"] == "TEST_JWT_CLIENT_ID"
+        assert claims["sub"] == "acct:example_username@TEST_AUTHORITY"
+        assert before <= claims["nbf"] <= after
+        assert claims["exp"] > before
+
+    def test_it_includes_the_group(self, pyramid_request, config):
+        groups = config["services"][0]["groups"]
+
+        assert groups == ["example_groupid"]
+
+    def test_it_is_empty_if_provisioning_feature_is_disabled(
+        self, context, pyramid_request
+    ):
+        context.provisioning_enabled = False
+
+        assert JSConfig(context, pyramid_request).config["hypothesisClient"] == {}
+
+    def test_it_is_mutable(self, config):
+        config.update({"a_key": "a_value"})
+
+        assert config["a_key"] == "a_value"
+
+    @pytest.fixture
+    def config(self, config):
+        return config["hypothesisClient"]
+
+
+class TestJSConfigRPCServer:
+    """Unit tests for the "rpcServer" sub-dict of JSConfig."""
+
+    def test_it(self, config):
+        assert config == {"allowedOrigins": ["http://localhost:5000"]}
+
+    @pytest.fixture
+    def config(self, config):
+        return config["rpcServer"]
+
+
+class TestJSConfigURLs:
+    """Unit tests for the "urls" sub-dict of JSConfig."""
+
+    def test_it(self, config):
+        assert config == {}
+
+    @pytest.fixture
+    def config(self, config):
+        return config["urls"]
+
+
+@pytest.fixture(autouse=True)
+def BearerTokenSchema(patch):
+    return patch("lms.resources._js_config.BearerTokenSchema")
+
+
+@pytest.fixture
+def bearer_token_schema(BearerTokenSchema):
+    return BearerTokenSchema.return_value
```

Comment:
Both of the bearer token fixtures here are only needed in `TestJSConfigAuthToken`. Moving them there would make that class easier to understand and prevent the large number of ambient fixtures we often accumulate at the bottom of test classes.

I've seen large collections of fixtures at the bottom of the file make it a pain to refactor when you need to split up tests.

Reply:
Just following my pattern again of always making import patches module-level fixtures. I'm gonna stick to that one as I've seen it go confusingly wrong too many times otherwise. In all other cases I'll put fixtures locally in test classes unless they're actually used by more than one class. The `bearer_token_schema` one could move but it's very related to `BearerTokenSchema`